crates/bevy_ecs/src/schedule_v3/schedule.rs

```diff
@@ -0,0 +1,1004 @@
+        self.graph.set_default_set(set);
+    }
+
+    pub fn set_executor(&mut self, executor: ExecutorKind) {
+        self.executor = match executor {
+            ExecutorKind::Simple => Box::new(SimpleExecutor::new()),
+            ExecutorKind::SingleThreaded => Box::new(SingleThreadedExecutor::new()),
+            ExecutorKind::MultiThreaded => Box::new(MultiThreadedExecutor::new()),
+        };
+    }
+
+    pub(crate) fn initialize(&mut self, world: &mut World) -> Result<(), BuildError> {
+        if self.graph.changed {
+            self.graph.initialize(world)?;
+            self.graph.update_schedule(&mut self.executable)?;
+            self.executor.init(&self.executable);
+        }
+
+        Ok(())
+    }
+
+    pub fn run(&mut self, world: &mut World) {
+        self.initialize(world).unwrap();
+        self.executor.run(&mut self.executable, world);
+    }
+
+    /// Iterates all component change ticks and clamps any older than [`MAX_CHANGE_AGE`](crate::change_detection::MAX_CHANGE_AGE).
+    /// This prevents overflow and thus prevents false positives.
+    ///
+    /// **Note:** Does nothing if the [`World`] counter has not been incremented at least [`CHECK_TICK_THRESHOLD`](crate::change_detection::CHECK_TICK_THRESHOLD)
+    /// times since the previous pass.
+    #[doc(hidden)]
+    pub fn check_change_ticks(&mut self, change_tick: u32, last_change_tick: u32) {
+        #[cfg(feature = "trace")]
+        let _span = bevy_utils::tracing::info_span!("check schedule ticks").entered();
+        todo!();
+    }
+}
+
+/// TBD
+#[derive(Resource)]
+pub struct Schedules {
+    inner: HashMap<BoxedScheduleLabel, Option<Schedule>>,
+    default: Option<BoxedScheduleLabel>,
+}
+
+impl Schedules {
+    pub fn new() -> Self {
+        Self {
+            inner: HashMap::new(),
+            default: None,
+        }
+    }
+
+    pub fn set_default(&mut self, label: impl ScheduleLabel) {
+        self.default = Some(label.dyn_clone());
+    }
+
+    pub fn insert(
+        &mut self,
+        label: impl ScheduleLabel,
+        schedule: Schedule,
+    ) -> Result<(), InsertError> {
+        let label = label.dyn_clone();
+        if self.inner.contains_key(&label) {
+            return Err(InsertError::AlreadyExists);
+        }
+        self.inner.insert(label, Some(schedule));
+        Ok(())
+    }
+
+    pub(crate) fn remove(&mut self, label: &dyn ScheduleLabel) -> Option<Schedule> {
+        self.inner.remove(&label.dyn_clone()).flatten()
+    }
+
+    pub fn check_out(&mut self, label: &dyn ScheduleLabel) -> Result<Schedule, CheckOutError> {
+        let label = label.dyn_clone();
+        match self.inner.get_mut(&label) {
+            Some(container) => match container.take() {
+                Some(schedule) => Ok(schedule),
+                None => Err(CheckOutError::AlreadyExtracted),
+            },
+            None => Err(CheckOutError::NotFound),
+        }
+    }
+
+    pub fn check_in(
```

Comment:
Please add docs to this. The naming doesn't tell the user exactly what this does.